### PR TITLE
Refactor : move retry on regional endpoint to SendDecorator

### PR DESCRIPTION
**Pull request by MartinForReal**
Signed-off-by: MartinForReal <fanshangxiang@gmail.com>

#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:
Refactor armclient and move retry on regional endpoint to SendDecorator

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

File: pkg/azureclients/armclient/azure_armclient.go [hunk 1]
```diff
@@ -151,15 +154,77 @@ func NormalizeAzureRegion(name string) string {
 	return strings.ToLower(region)
 }
 
+// DoExponentialBackoffRetry represents an autorest.SendDecorator with backoff retry.
```
Comment (nilo19) [inline pkg/azureclients/armclient/azure_armclient.go:157]: nit: modify the comment

**Review by feiskyer — COMMENTED** [review]
/lgtm
/approve

**Pull request merged**